### PR TITLE
Login funcitionality created

### PR DESCRIPTION
**Pull request by LuisFernandoSFonseca**
Login functionality created, Home screen xml created but not developed yet.

### CODE REVIEW

File: app/src/main/java/com/example/b2blearning/iLoginConfig.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import retrofit2.http.POST;
+import retrofit2.http.Query;
+
+
+//Creating interface to communicate with server.
+public interface iLoginConfig {
+    @POST("/api/users/login")
+    Call<User> directions(@Query("email") String email, @Query("password") String password);
+    //Call<User>getUser(@Path("login?email=cicrano@interncommerce.com&password=654321")String user);
```
Comment (xandebianchi) [inline app/src/main/java/com/example/b2blearning/iLoginConfig.java:15]: Remove unused comments from code.
@LuisFernandoSFonseca

File: app/src/main/java/com/example/b2blearning/Utility.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.example.b2blearning;
+
+import android.content.Context;
+import android.widget.Toast;
+
+public class Utility {
+    public static boolean loginCheck(String email, String password, Context context) {
+        boolean check = false;
+        if (email.equals("")) {
+            Toast.makeText(context, "Please submit an email", Toast.LENGTH_LONG).show();
```
Comment (xandebianchi) [inline app/src/main/java/com/example/b2blearning/Utility.java:10]: Rephrase to "Please inform an email"
@LuisFernandoSFonseca

File: app/src/main/java/com/example/b2blearning/Utility.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        }
+        else if (!email.contains("@") && !email.contains(".com")){
+            Toast.makeText(context, "Use a valid e-mail such as:\n example@email.com", Toast.LENGTH_LONG).show();
+        }
+        else if (password.equals("")) {
+            Toast.makeText(context, "Please submit a password", Toast.LENGTH_LONG).show();
```
Comment (xandebianchi) [inline app/src/main/java/com/example/b2blearning/Utility.java:16]: Rephrase to "Please inform an password"
@LuisFernandoSFonseca

File: app/src/main/java/com/example/b2blearning/MainActivity.java [hunk 4]
```diff
@@ -5,22 +5,65 @@
...
+                            } else {
+                                Toast.makeText(getBaseContext(), "Login response : Error " + code + ", incorrect credentials", Toast.LENGTH_LONG).show();
+                            }
+
+                        }
+
+                        @Override
+                        public void onFailure(Call<User> call, Throwable t) {
+                            Toast.makeText(getBaseContext(), "Error " + t.getMessage(), Toast.LENGTH_LONG).show();
+
```
Comment (xandebianchi) [inline app/src/main/java/com/example/b2blearning/MainActivity.java:59]: Remove line with space.
@LuisFernandoSFonseca

File: app/src/main/java/com/example/b2blearning/MainActivity.java [hunk 5]
```diff
@@ -5,22 +5,65 @@
...
+                    login.directions(email, password).enqueue(new Callback<User>() {
+                        @Override
+                        public void onResponse(Call<User> call, Response<User> response) {
+                            int code = response.code();
+                            if (code == 200) {
+                                Toast.makeText(getBaseContext(), "Login response " + code + ": Successful", Toast.LENGTH_LONG).show();
+                            } else {
+                                Toast.makeText(getBaseContext(), "Login response : Error " + code + ", incorrect credentials", Toast.LENGTH_LONG).show();
+                            }
+
```
Comment (xandebianchi) [inline app/src/main/java/com/example/b2blearning/MainActivity.java:53]: Remove line with space
@LuisFernandoSFonseca

File: app/src/main/java/com/example/b2blearning/MainActivity.java [hunk 6]
```diff
@@ -5,22 +5,65 @@
...
+    private boolean check;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        emailView = findViewById(R.id.login);
+        passwordView = findViewById(R.id.password);
+
```
Comment (xandebianchi) [inline app/src/main/java/com/example/b2blearning/MainActivity.java:28]: Remove line with space.
@LuisFernandoSFonseca

File: app/src/main/java/com/example/b2blearning/MainActivity.java [hunk 7]
```diff
@@ -5,22 +5,65 @@
...
+                if(check){
+                    login.directions(email, password).enqueue(new Callback<User>() {
+                        @Override
+                        public void onResponse(Call<User> call, Response<User> response) {
+                            int code = response.code();
+                            if (code == 200) {
+                                Toast.makeText(getBaseContext(), "Login response " + code + ": Successful", Toast.LENGTH_LONG).show();
+                            } else {
+                                Toast.makeText(getBaseContext(), "Login response : Error " + code + ", incorrect credentials", Toast.LENGTH_LONG).show();
```
Comment (xandebianchi) [inline app/src/main/java/com/example/b2blearning/MainActivity.java:51]: Verify if all other codes returned are related to "incorrect credentials".
If not, divide the answers in another possibilities.
@LuisFernandoSFonseca

**Review by xandebianchi — COMMENTED** [review]
Verify comments.

File: app/src/main/java/com/example/b2blearning/MainActivity.java (hunk 4 above)
Comment (LuisFernandoSFonseca) [reply app/src/main/java/com/example/b2blearning/MainActivity.java:59]: Will resolve on next PR

File: app/src/main/java/com/example/b2blearning/MainActivity.java (hunk 5 above)
Comment (LuisFernandoSFonseca) [reply app/src/main/java/com/example/b2blearning/MainActivity.java:53]: Will resolve on next PR

File: app/src/main/java/com/example/b2blearning/MainActivity.java (hunk 6 above)
Comment (LuisFernandoSFonseca) [reply app/src/main/java/com/example/b2blearning/MainActivity.java:28]: Will resolve on next PR

File: app/src/main/java/com/example/b2blearning/MainActivity.java (hunk 7 above)
Comment (LuisFernandoSFonseca) [reply app/src/main/java/com/example/b2blearning/MainActivity.java:51]: The only error code the server return is 401(Unauthorized)

**Pull request merged**